### PR TITLE
Skip identity operations in OneStagePlan.

### PR DESCRIPTION
**Pull request by Wosin**
Still in progress. Need to figure out how to prevent columns from getting cleaned and why the results are not correct.

### CODE REVIEW

File: src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala [hunk 1]
```diff
@@ -50,6 +53,16 @@ final case class OneStageEvaluationPlan(
 
   override def output: Seq[Attribute] = outputExpressions.map(_.toAttribute)
 
+  private val isProject = veFunction.functionName.startsWith("project")
```
Comment (wgip) [inline src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala:56]: I'd create a new plan with `Project` in its name.

File: src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala [hunk 2]
```diff
@@ -67,12 +80,36 @@ final case class OneStageEvaluationPlan(
               cols = veColBatch.cols,
               results = veFunction.results
             )
-
-            val outBatch = VeColBatch.fromList(cols)
+            val outBatch = if(isProject) {
+             val outputColumns = outputExpressions
+                .foldLeft((0, 0, Seq.empty[VeColVector])) {
```
Comment (wgip) [inline src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala:85]: Might make sense to add unit tests for this logic

File: src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala [hunk 3]
```diff
@@ -50,6 +53,16 @@ final case class OneStageEvaluationPlan(
...
 
+  private val isProject = veFunction.functionName.startsWith("project")
+
+  private val copiedRefs = outputExpressions.filter(_.isInstanceOf[AttributeReference])
+  private val idsToCopy = child
+    .outputSet
+    .toList
+    .zipWithIndex
+    .collect {
+      case (ref, idx) if(copiedRefs.find(find => ref.name == find.name).isDefined) => idx
```
Comment (wgip) [inline src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala:64]: should get unit tests here ,including some negative ones; i see this potentially filtering successfully although it shouldn't

File: src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala (hunk 1 above)
Comment (Wosin) [reply src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala:56]: Done.

File: src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala (hunk 3 above)
Comment (Wosin) [reply src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala:64]: Sure, gonna write smth. But not really usre if there is any other way to compare those attrs.

File: src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala [hunk 4]
```diff
@@ -78,6 +78,16 @@ final case class OneStageEvaluationPlan(
 
   override def output: Seq[Attribute] = outputExpressions.map(_.toAttribute)
 
+  private val isProject = veFunction.functionName.startsWith("project")
```
Comment (wgip) [inline src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala:81]: Can remove these now

Comment (Wosin) [reply src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala:81]: Yeah, somehow this got messed up during rebas.e

**Pull request merged**